src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccessTokenServiceImpl.java

```diff
@@ -61,12 +122,121 @@ protected ClientRequest buildRefreshAccessTokenClientRequest() {
                     + CUKFSConstants.AMPERSAND + ConcurConstants.CLIENT_ID_URL_PARAM + CUKFSConstants.EQUALS_SIGN + getConsumerKey()
                     + CUKFSConstants.AMPERSAND + ConcurConstants.CLIENT_SECRET_URL_PARAM + CUKFSConstants.EQUALS_SIGN + getSecretKey());
         } catch (URISyntaxException e) {
+            LOG.error("buildRefreshAccessTokenClientRequest, there was a problem building refresh access token client request.", e);
             throw new RuntimeException("An error occured while building the refresh access token URI: ", e);
         }
-        
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("buildRefreshAccessTokenClientRequest, URI: " + uri);
+        }
         return builder.build(uri, HttpMethod.GET);
     }
 
+    @Transactional
+    @Override
+    public void revokeAndReplaceAccessToken() {
+        boolean success = executeRevokeAccessTokenRequest();
+        
+        if (success) {
+            requestNewAccessToken();
+        } else {
+            throw new RuntimeException("Error revoking access token: Unsuccessful response from Concur");
+        }
+    }
+    
+    @Transactional
+    @Override
+    public void revokeAccessToken() {
+        boolean success = executeRevokeAccessTokenRequest();
+        if (success) {
+            setWebserivceCredentialValues(StringUtils.EMPTY, StringUtils.EMPTY, StringUtils.EMPTY);
+        } else {
+            throw new RuntimeException("Error revoking access token: Unsuccessful response from Concur");
+        }
+    }
+    
+    @Override
+    public void deleteTokensFromDatabase() {
```

Comment:
I think this method's name is misleading because it's not really deleting the entries from the database but rather reseting all values to empty